File: src/main/java/etlmail/front/gui/helper/DocumentAdapter.java
package etlmail.front.gui.helper;

import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;

public abstract class DocumentAdapter implements DocumentListener {
	@Override
	public final void removeUpdate(DocumentEvent e) {
		update(e);
	}

	@Override
	public final void insertUpdate(DocumentEvent e) {
		update(e);
	}

	@Override
	public final void changedUpdate(DocumentEvent e) {
		update(e);
	}

	private void update(DocumentEvent e) {
		update(ModelUtils.getText(e.getDocument()));
	}

	protected abstract void update(String newText);
}

File: src/main/scala/etlmail/front/gui/helper/DocumentAdapter.scala
package etlmail.front.gui.helper

import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

abstract class DocumentAdapter extends DocumentListener {
  final def removeUpdate(e: DocumentEvent) {
    update(e)
  }

  final def insertUpdate(e: DocumentEvent) {
    update(e)
  }

  final def changedUpdate(e: DocumentEvent) {
    update(e)
  }

  def update(e: DocumentEvent) {
    update(ModelUtils.getText(e.getDocument))
  }

  protected def update(newText: String)
}
